descartes_light/include/descartes_light/interface/edge_evaluator.h

```diff
@@ -32,13 +32,59 @@ template <typename FloatType>
 class EdgeEvaluator
 {
 public:
+  using Ptr = typename std::shared_ptr<EdgeEvaluator<FloatType>>;
+  EdgeEvaluator(std::size_t dof) : dof_(dof) {}
   virtual ~EdgeEvaluator() {}
 
   virtual bool evaluate(const Rung_<FloatType>& from,
                         const Rung_<FloatType>& to,
-                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges) = 0;
+                        std::vector<typename LadderGraph<FloatType>::EdgeList>& edges)
+  {
+    const auto n_start = from.data.size() / dof_;
+    const auto n_end = to.data.size() / dof_;
 
-  typedef typename std::shared_ptr<EdgeEvaluator<FloatType>> Ptr;
+    // Allocate if needed because of Compound Evaluator)
+    if (edges.size() == 0)
+      edges.resize(n_start);
+
+    assert(edges.size() == n_start);
```

Comment:
You could create a compound edge evaluator that calls evaluate instead of consider edge so you do not want to resize which would nullify the data already in the vector. Should only resize if zero, but I have added the assert to catch if something goes wrong, the assert should not ever be hit but just encase something changes upstream or a user defined edge evaluator inadvertently changes the size to something incorrect.

Reply:
I am not sure if this would ever be done but if we want to fix the size then I would remove the virtual.

Reply:
This is why I want to change this function signature: it's unclear what state `edges` should be in when you call this function and then you need to add checks on `edges` in this function to accommodate all the different ways it could come in. I think the function signature change described below would make this better

Reply:
@marip8 You good with me removing the virtual and doing as you recommended?

Reply:
I'm on-board with changing this function

Reply:
Done.